react/best-practices.md

```diff
@@ -0,0 +1,35 @@
+# React / Best practices
+
+### Hooks
+
+#### Return Object from custom hooks instead of Array
+
+GOOD
+
+```js
+export const useData = () => {
+  const [someData, setSomeData] = useState();
+  const [someOtherData, setSomeOtherData] = useState();
+
+  return {
+    someData,
+    someOtherData,
+  };
+};
+```
+
+BAD
+
+This way when using the hook, the order of variables would matter.
```

Comment:
Yep, that's right. On the other hand, using the object-approach means we're dependent on the names, aren't we?

Reply:
You can choose to rename those explicitly. I think this approach is way less error-prone.

Reply:
> You can choose to rename those explicitly.

You mean the example below, right?

```js
const [someData: foo, someOtherData: bar] = useData();
```

Reply:
Not sure :) If this works, that's super easy, worst case you can re-assign separately.

Reply:
Sorry, that should be enclosed with `{}`.

Reply:
Yep, it indeed works.

https://codesandbox.io/s/custom-hooks-object-o8oxw

Reply:
Thanks. Added some usage examples: https://github.com/c-hive/guides/pull/35/commits/0cbe04d4a250b6651af1f9c3ca1f5a1f0cbf6f81